Replace the WDL language grammar

claymcleod opened the pull request:
This PR updates support for the WDL to the latest version ([v1.2](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md)).

## Description

We've written the grammar from scratch to be compliant with the latest version of WDL—currently, the grammar used in linguist is around 6 years old (and represents a very old version of WDL).

A list of some (but not all) improvements being:

- Added all keywords (e.g., `requirements`).
- Shell highlighting within `command` blocks.
- Better semantic classification of different keywords.

## Checklist:

- [x] **I am changing the source of a syntax highlighting grammar**
  - Old: https://github.com/broadinstitute/wdl-sublime-syntax-highlighter
  - New: https://github.com/stjude-rust-labs/sprocket-vscode

_(From the **I am adding a new language of the template.** section, as I'm adding a new example)_
- [x] I have included a real-world usage sample for all extensions added in this PR:
  - Sample source(s): 
    - [`fq.wdl`](https://github.com/stjudecloud/workflows/blob/833b97fb582fe7d4f6df9d29a645d0289f1fc92e/tools/fq.wdl)
  - Sample license(s): MIT

lildude reviewed (changes requested):
Please do not change the samples. They are used to train the classifier and there will almost always be files on GitHub using older syntax.

If you want to help detect newer syntax, add more samples.

claymcleod commented:
Okay, I've reverted those samples and added a new one! Let me know what you think, @lildude.

H200a commented:
> This PR updates support for the WDL to the latest version ([v1.2](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md)).
> 
> 
> 
> ## Description
> 
> 
> 
> We've written the grammar from scratch to be compliant with the latest version of WDL—currently, the grammar used in linguist is around 6 years old (and represents a very old version of WDL).
> 
> 
> 
> A list of some (but not all) improvements being:
> 
> 
> 
> - Added all keywords (e.g., `requirements`).
> 
> - Shell highlighting within `command` blocks.
> 
> - Better semantic classification of different keywords.
> 
> 
> 
> ## Checklist:
> 
> 
> 
> - [x] **I am changing the source of a syntax highlighting grammar**
> 
>   - Old: https://github.com/broadinstitute/wdl-sublime-syntax-highlighter
> 
>   - New: https://github.com/stjude-rust-labs/sprocket-vscode
> 
> 
> 
>

H200a commented:
‫أُرسلت من الـ iPhone‬‫في ٣٠‏/٠٧‏/٢٠٢٤ الساعة ٨:٥٢ م، كتب/كتبت ‏Clay McLeod ***@***.***>:‬﻿
Okay, I've reverted those samples and added a new one! Let me know what you think, @lildude.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

lildude commented on lib/linguist/heuristics.yml:
```diff
@@ -825,6 +825,10 @@ disambiguations:
+- extensions: ['.wdl']
+  rules:
+  - language: WDL
+    pattern: '^version'
```
Why are you adding this? This change adds no value as this is the only language that uses this extension so the heuristic will never be used.

This PR is about changing the grammar so no other changes are needed.

claymcleod replied:
Yeah, I see now. I saw mention of updating the heuristics in the PR template, and, even though it was in the section for a new package, I figured it was required and was simply missed because the WDL language hadn't been updated in so long. That said, the wording does imply that it's only needed to help delineate between packages of the same extension.

I've removed this.

lildude reviewed (changes requested):
Please also replace your sample with a real world example and update the template to state where it comes from and the license of the sample.

claymcleod commented:
> Please also replace your sample with a real world example and update the template to state where it comes from and the license of the sample.

This should be addressed: I added a new, real-world example and also added the license to the template for the PR and the header of that sample.

lildude commented:
The test failure suggests you might not have used the documented method of replacing the grammar as the script sorts the submodules.

Please use the script as documented. You can replace the grammar with itself using the script.

claymcleod commented:
Okay, I think I've resolved this now—I had used the script, but had done some spelunking on the codebase beforehand to try to understand how it worked (in unstaged changes).

I checked out a fresh version of the repo, applied my changes, and then got `bundle exec rake` running with no errors.
